### DISCUSSION TITLE
Github actions multi repository checkout

### DISCUSSION THREAD

**Discussion by grajeshkannan** [community/community#27030] [Actions]
<p>Hi there,</p>
<p>My project uses microservices architecture and each microservice is kept in a seperate repo.<br>
But there is a container repo, which offers a foundation to share config files etc and to build, test all microservices together as a system.</p>
<p>With github actions, if i use ‘actions/checkout@v2’ to download the base repo in one step and download a microservice repo in the next step, it deletes the previously downloaded repo and then replace it with this, is there a way i can download multip repos in sequence and able to see all the source code together to build the final system - because of the dependency in nature ?</p>
<p>best regards<br>
Raj</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can do this by specifying different directories for the different directories using the <code>path</code> parameter. See the example <a href="https://github.com/actions/checkout#Checkout-multiple-repos-side-by-side">Checkout multiple repos (side by side)</a> in the actions/checkout Readme.</p>

**Comment by grajeshkannan** [comment]
<p>Thanks <a class="mention" href="/airtower-luna">@airtower-luna</a>, it allowed me to get both repo at the same time like below<br>
/home/runner/work/repo1<br>
/home/runner/work/repo2</p>
<p>but the restriction in path is, it needs to be under the “/home/runner/work” is there a way i can checkout the repo to the go src directory ?<br>
/home/runner/go/src/github.com/xx/repo1<br>
/home/runner/go/src/github.com/xx/repo1/services/repo2 ?</p>

**Comment by Novodes** [comment]
<p>I wanted to add that you can do it side by side in multiple repos as mentioned in the link you provided but the downside is that you have to indicate the list of repos you want to clone in a serial manner in the yml file. In my case, I wanted to use a script I had to clone the repos I wanted (which I had listed in a txt file in my main repo). In that case, I couldn’t use the yml file to clone the repos because I didn’t know what they were at that time.</p>
<p>So instead, I used my personal access token this way with a bash script: <a href="https://stackoverflow.com/questions/68255301/how-to-clone-multiple-private-repositories-using-github-actions" rel="noopener nofollow ugc">bash - How to clone multiple private repositories using GitHub Actions? - Stack Overflow</a></p>
